pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh

```diff
@@ -0,0 +1,22 @@
+yarnConfigureHook(){
+    runHook preConfigure
+    echo "Executing yarnConfigureHook"
+
+    # Use a constant HOME directory
+    export HOME=/tmp/home
+    # TODO: Maybe support other shell variables such as $yarnOfflineCache?
+    yarn config --offline set yarn-offline-mirror $offlineCache
+    fixup-yarn-lock yarn.lock
+    yarn install \
+        --offline --frozen-lockfile \
+        --force --production=false \
+        --ignore-engines --ignore-scripts
+    patchShebangs node_modules
```

Comment:
While we are reworking things. We cannot limit this to node_modules/.bin?

Reply:
or a pattern like node_modules/*/bin. They seem to be all over the place...

Reply:
> or a pattern like node_modules/*/bin. They seem to be all over the place...

It's worth checking in anycase whether this `patchShebangs` is really needed. I'll run a `nixpkgs-review` on a strong machine with and without it when I finish iterating all packages.

Reply:
Uh, that would be super awesome if we can avoid carrying this forward with us.

Reply:
Similarly to my comment to @viraptor [here](https://github.com/NixOS/nixpkgs/pull/318015#discussion_r1664011235), I think it should be out of scope for this PR to remove that `patchShebangs` command, since this is also the behavior of all of these packages before this PR.

This also helps me [demonstrate in the little `diff --recursive` shell loop below](https://github.com/NixOS/nixpkgs/pull/318015#issuecomment-2206765706) how the packages barely change due to this PR. I still kept the TODO comment near there though - maybe someone will be bothered by this in the future and provide a clean solution.